src/easier68k/core/models/list_file.py

```diff
@@ -138,14 +158,93 @@ def load_from_json(self, json_str: str):
         loaded = json.loads(json_str)
         self.symbols = loaded['symbols']
         self.data = loaded['data']
+        self.starting_execution_address = loaded['startingExecutionAddress']
+
+    def read_s_record_filename(self, filepath: str):
+        """
+        Read the S record at the given file path
+        :param filepath: Path to an S record
+        :return:
+        """
+        with open(filepath, 'r') as f:
+            for line in f:
+                # process the line in the file
+                self.__process_s_record_line(line)
+
+    def __process_s_record_line(self, line: str):
+        """
+        Process a single line of the S record
+        This is defined here: http://www.easy68k.com/easy68ksrecord.htm
+        :param line:
+        :return:
+        """
+        # type of record
+        record_type = SRecordType.parse(line[:2])
+        # count of remaining character pairs in the record
+        count = int(line[2:4], 16)
+        # address 2 3 or 4 bytes as hex
+        address_val_len = 0
+
+        if record_type is SRecordType.S0:
+            # address field is unused
+            address_val_len = 4
+        elif record_type is SRecordType.S1:
+            # 2 bytes
+            address_val_len = 4
+        elif record_type is SRecordType.S2:
+            # 3 bytes
+            address_val_len = 6
+        elif record_type is SRecordType.S3:
+            # 4 bytes
+            address_val_len = 8
+        elif record_type is SRecordType.S5:
+            # the address field is interpreted as a 2 byte value
+            # and contains the counts of S1 2 and 3 records prev
+            # transmitted
+            address_val_len = 4
+        elif record_type is SRecordType.S7:
+            # termination record
+            # contains the starting execution address
+            # as 4 bytes
+            address_val_len = 8
+        elif record_type is SRecordType.S8:
+            # termination record
+            # contains the starting execution address
+            # as 3 bytes
+            address_val_len = 6
+        elif record_type is SRecordType.S9:
+            # termination record
+            # contains the starting execution address
+            # as 2 bytes
+            address_val_len = 4
+
+        # get the address for the data
+        address = int(line[4:4+address_val_len], 16)
+
+        # get the data that should be between 0-64 characters
+        data = line[4+address_val_len:-2]
+
+        # the last two characters as a hexadecimal value
+        # with the least significant byte of the ones complement of the sum
+        # of the byte values represented by the pairs of characters
+        # making up the count, address and data pairs
+
+        # for now, I don't care about the checksum
+        checksum = line[-2:]
+
+        if record_type in [SRecordType.S1, SRecordType.S2, SRecordType.S3]:

```

Comment:
Okay, I need to do if statements like this more often, this is a nice and slick way